Define that sync objects do not become signaled until control returns to the event loop.

kenrussell opened the pull request:
Follow the same semantics for sync objects as were previously used for
queries. Add a test for the new behavior.

kenrussell commented:
@jdashg please review.

@zhenyao @kainino0x @jdarpinian FYI.

kainino0x commented:
Tests LGTM as well.

jdashg commented on sdk/tests/conformance2/sync/sync-webgl-specific.html:
```diff
@@ -77,11 +78,99 @@
+  // Verify as best as possible that the implementation doesn't allow a sync
+  // object to become signaled in the same frame, by spin-looping for some time.
+  var startTime = Date.now();
+  while (Date.now() - startTime < 2000) {
```
300ms-1000ms is fine. 2000 seems unnecessarily high.

kenrussell replied:
Changed to 1000 ms.

jdashg commented on sdk/tests/conformance2/sync/sync-webgl-specific.html:
```diff
@@ -77,11 +78,99 @@
+  if (startTimeOfFinish == 0) {
+    startTimeOfFinish = Date.now();
+  }
+  if (Date.now() - startTimeOfFinish > 2000) {
```
2000 used without being a named constant again. Please name and define one.

kenrussell replied:
Thanks for your feedback. Named and defined one.

jdashg commented:
Ok, this works for me. The title here is inaccurate though, since this matches the query behavior that the app must return to the event loop, not necessarily the next frame. It's worth correcting so the changelog shows what we actually did.

kenrussell commented:
Thanks @jdashg for your review. I rebased, squashed and edited the commit message to be accurate. Will update the pull request title here too. Merging.
